File: weddingplanner/WeddingPlannerBackend/Entities/PlannerInvitation.cs
using System.ComponentModel.DataAnnotations;

namespace WeddingPlannerBackend.Entities;

public enum InvitationStatus
{
    PENDING,
    ACCEPTED,
    REJECTED,
    EXPIRED
}

public class PlannerInvitation
{
    [Key]
    public Guid Id { get; set; } = Guid.NewGuid();
    
    [Required]
    public Guid EventId { get; set; }
    public Event Event { get; set; } = null!;
    
    [Required]
    public Guid SenderId { get; set; }
    public User Sender { get; set; } = null!;
    
    [Required]
    public Guid ReceiverId { get; set; }
    public User Receiver { get; set; } = null!;
    
    public EventRole Role { get; set; } = EventRole.PLANNER;
    public InvitationStatus Status { get; set; } = InvitationStatus.PENDING;
    
    public string? Message { get; set; }
    
    public DateTime ExpiresAt { get; set; } = DateTime.UtcNow.AddDays(7);
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}

File: weddingplanner/packages/types/src/types/plannerInvitation.ts
import { BaseEntity } from "./baseEntity";
import { EventRole } from "./enums";

export interface PlannerInvitation extends BaseEntity {
  eventId: string;
  senderId: string;
  receiverId: string;
  role: EventRole | string; // Role being invited to
  status: string; // Will be one of the InvitationStatus values from Prisma
  message?: string | null;
  expiresAt: Date | string;
}

export interface PlannerInvitationWithRelations extends PlannerInvitation {
  event: {
    id: string;
    eventName: string;
    eventDescription?: string | null;
    eventType: string;
    eventDate?: Date | string | null;
  };
  sender: {
    id: string;
    name: string;
    email: string;
  };
  receiver: {
    id: string;
    name: string;
    email: string;
  };
}

export interface CreatePlannerInvitationRequest {
  eventId: string;
  receiverEmail: string;
  role: EventRole | string;
  message?: string;
}

export interface PlannerInvitationResponse {
  status: 'accept' | 'reject';
}